std-bits/table/src/main/java/org/enso/table/data/column/builder/object/InferredBuilder.java

```diff
@@ -107,22 +108,26 @@ private void initBuilderFor(Object o) {
     currentBuilder.appendNulls(currentSize);
   }
 
-  private record RetypeInfo(Class<?> clazz, int type) {}
+  private record RetypeInfo(Class<?> clazz, StorageType type) {}
 
   private static final List<RetypeInfo> retypePairs =
       List.of(
-          new RetypeInfo(Boolean.class, Storage.Type.BOOL),
-          new RetypeInfo(Long.class, Storage.Type.LONG),
-          new RetypeInfo(Double.class, Storage.Type.DOUBLE),
-          new RetypeInfo(String.class, Storage.Type.STRING),
-          new RetypeInfo(BigDecimal.class, Storage.Type.DOUBLE),
-          new RetypeInfo(LocalDate.class, Storage.Type.DATE),
-          new RetypeInfo(LocalTime.class, Storage.Type.TIME_OF_DAY),
-          new RetypeInfo(ZonedDateTime.class, Storage.Type.DATE_TIME),
-          new RetypeInfo(Float.class, Storage.Type.DOUBLE),
-          new RetypeInfo(Integer.class, Storage.Type.LONG),
-          new RetypeInfo(Short.class, Storage.Type.LONG),
-          new RetypeInfo(Byte.class, Storage.Type.LONG));
+          new RetypeInfo(Boolean.class, Constants.BOOLEAN),
+          new RetypeInfo(Long.class, Constants.INTEGER_64),
+          new RetypeInfo(Double.class, Constants.FLOAT_64),
+          new RetypeInfo(String.class, Constants.STRING),
+          // TODO [RW] I think BigDecimals should not be coerced to floats, we should add Decimal
+          // support to in-memory tables at some point
+          // new RetypeInfo(BigDecimal.class, StorageType.FLOAT_64),
```

Comment:
While I agree the todo - think the line should be enabled?

Reply:
Without it, `BigDecimal` will be just stored as an object and the column will get a `Mixed` type. Won't that be better until we get proper support?

Reply:
We don't have a dedicated storage type for `BigDecimal` anyway. Nor for `BigInteger` which is more problematic since we have a real "bug" here because BigIntegers are natively supported by Enso (an Enso `Integer` can be a `long` or `BigInteger`.

I'm not sure the retyping will work correctly if I just enable it, it would require additional work. Let's keep it as-is for now. If we want support for BigDecimal in Table, let's add a ticket and do it properly.

Reply:
For the record, this is what currently happens if I put a BigInteger into a Table - it's not bad, because since that type is unrecognized, it is just treated as a `Mixed` column:

![image](https://user-images.githubusercontent.com/1436948/229095646-acc09361-dcf4-4b65-a770-117e1f69095f.png)

Not ideal, because we cannot do arithmetic operations on it directly (since it is not of a numeric type). But at least there's no data corruption.